pcdsdevices/ipm.py

```diff
@@ -49,15 +96,276 @@ def removed(self):
 
-    def remove(self, moved_cb=None, timeout=None, wait=False):
-        """Moves the target out of the beam. Diode never blocks."""
-        return self.target.remove(moved_cb=moved_cb,
-                                  timeout=timeout,
-                                  wait=wait)
-
-    remove.__doc__ += insert_remove
-
     @property
     def transmission(self):
         """Returns the target's transmission value. Diode never blocks."""
         return self.target.transmission
+
+
+class IPIMBChannel(Device):
+    """
+    Class for a single channel read out by an ipimb box
+
+    Parameters
+    ----------
+    prefix : ``str``
+        Ipimb base PV
+
+    name : ``str``
+        Alias for the ipimb box
+
+    channnel_index : ``int``
+        Index for gauge (0-3)
+    """
+    amplitude = FCpt(EpicsSignal, '{self.prefix}:CH{self.channel_index}',
+                     kind='hinted')
+    gain = FCpt(EpicsSignal,
+                '{self.prefix}:ChargeAmpRangeCH{self.channel_index}',
+                kind='config', string=True)
+
+    base = FCpt(EpicsSignal, '{self.prefix}:CH{self.channel_index}_BASE',
+                kind='config')
+    scale = FCpt(EpicsSignal, '{self.prefix}:CH{self.channel_index}_SCALE',
+                 kind='config')
+
+    def __init__(self, prefix, *, name, channel_index,  **kwargs):
+        self.channel_index = channel_index
+        super().__init__(prefix, name=name, **kwargs)
+
+
+class IPIMB(Device):
+    """
+    Class for an ipimb box.
+
+    Parameters
+    ----------
+    prefix : ``str``
+        Ipimb base PV
+
+    name : ``str``
+        Alias for the ipimb
+
+    prefix_ioc : ``str``
+        Ipimb base PV for IOC PVs
+
+    Components - readback:
+    total sum of all 4 channels (i0 if stadard IPM device)
+    x&y beam position position calculated from 4 channels
+    Components - configuration:
+    bias: voltage aplied to diodes in V
+    delay: delay of trigger relative to input EVR
+    trigger:
+    """
+    isum = Cpt(EpicsSignal, ':SUM', kind='hinted')
+    xpos = Cpt(EpicsSignal, ':XPOS', kind='normal')
+    ypos = Cpt(EpicsSignal, ':YPOS', kind='normal')
+    evr_channel = Cpt(Trigger, ':TRIG:TRIG0', kind='normal')
+    delay = Cpt(EpicsSignal, ':TrigDelay', kind='config')
+    bias = Cpt(EpicsSignal, ':DiodeBias', kind='config')
+    ch0 = FCpt(IPIMBChannel, '{self.prefix}', channel_index=0, kind='normal')
+    ch1 = FCpt(IPIMBChannel, '{self.prefix}', channel_index=1, kind='normal')
+    ch2 = FCpt(IPIMBChannel, '{self.prefix}', channel_index=2, kind='normal')
+    ch3 = FCpt(IPIMBChannel, '{self.prefix}', channel_index=3, kind='normal')
+
+    # also be careful that you have to wait until trigger is enabled
+    # again before device can be used again.
+    # wait functionality?
+
+    def __init__(self, prefix, *, name, prefix_ioc, **kwargs):
+        if not prefix_ioc:
+            self._prefix_ioc = 'IOC:%s' % prefix
+        else:
+            self._prefix_ioc = prefix_ioc
```

Comment:
I've been considering doing something like this on all devices so we can quickly check ioc status or reboot from the python sessions. :+1: different use here but still good.